torch/optim/lr_scheduler.py

```diff
@@ -0,0 +1,206 @@
+import numpy as np
+from bisect import bisect_right
+from torch.optim.optimizer import Optimizer
+
+
+class LambdaLR(object):
```

Comment:
I believe that this part is less complicated than the optimizer. Also, the step() function should be called before each epoch. (or after for ReduceLROnPlateau) This is also different from optimizer.